vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/IntroduceCollectionRule.java

```diff
@@ -64,7 +64,8 @@
     @Override
     public boolean rewritePre(Mutable<ILogicalOperator> opRef, IOptimizationContext context) {
         VXQueryOptimizationContext vxqueryContext = (VXQueryOptimizationContext) context;
-        String args[] = getCollectionName(opRef);
+        String args[] = getFunctionalArguments(opRef, BuiltinFunctions.FN_COLLECTION_WITH_TAG_2, BuiltinFunctions
+                .FN_COLLECTION_1);
```

Comment:
It's bit unclear to me creating set of FunctionalIdentifires. Is it a single set of all identifres related to indexing and collection or separate ones.